tests/derive/groups.rs

```diff
@@ -120,3 +120,47 @@ fn helpful_panic_on_duplicate_groups() {
     use clap::CommandFactory;
     Opt::command().debug_assert();
 }
+
+#[test]
+fn required_group() {
```

Comment:
Tests are where I was the most unsure about what to do. If there's a better/simpler way to structure these, I'm all ears.